### ISSUE TITLE
Upgrade JSON-path to 1.2.0 to align with Spring 4.1.x

### ISSUE THREAD

**Issue by aheusingfeld** [spring-projects/spring-hateoas#315]
Spring 4.1.x upgraded to json-path 1.2.0 with https://jira.spring.io/browse/SPR-12299, I guess that Spring HATEOAS should follow the lead.

**Comment by olivergierke** [comment]
The reason we didn't upgrade to a release better than 0.9.1 is that 1.0 contains breaking changes in the APIs. In fact, [this commit](https://github.com/spring-projects/spring-hateoas/commit/de59fa40b932f398f8281b1810b5c0209851e031) adds the necessary reflective workaround to work with bot 0.9 and 1.x versions of the JSON Path library. 

That's by the way the very same workaround Spring Framework uses to workaround the API changes (see the comments in the ticket you linked), too, and thus also stays compatible with 0.9. We just decided not to move to the newer dependency at build time as Spring HATEOAS is is used in the Spring Data Evans release train, which is in maintenance mode and we can't upgrade to an API-breaking third-party dependency in service releases.

Long story short: Spring HATEOAS works fine on 0.9 as well as 1.x versions of JSON Path just as Spring Framework does. Or did you actually run into an issue with the new version?

**Comment by aheusingfeld** [comment]
Yes. I ran exactly into the exception mention in https://jira.spring.io/browse/SPR-12299?focusedCommentId=109040&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-109040 when working with https://github.com/spring-projects/spring-hateoas/issues/314 but haven't been able to reproduce the root-cause, yet, due to higher priorities. Will come back to it as soon as I find the time!

**Comment by olivergierke** [comment]
Did you use 0.17?

**Comment by aheusingfeld** [comment]
No, I'm using
```
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-hateoas</artifactId>
    </dependency>
```
which is still on 0.16

**Comment by olivergierke** [comment]
You should see this working when upgrading to 0.17. I could verify #314 still broken but are about to push a fix for that.

**Comment by aheusingfeld** [comment]
Nice! Both fixes verified! I just added the following to my pom.xml:
```
  <properties>
    <jsonpath.version>1.2.0</jsonpath.version>
    <spring-hateoas.version>0.17.0.RELEASE</spring-hateoas.version>
  </properties>
```
Thanks! :+1:
